irohad/ametsuchi/impl/failover_callback.hpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_FAILOVER_CALLBACK_HPP
+#define IROHA_FAILOVER_CALLBACK_HPP
+
+#include <memory>
```

Comment:
There is a problem - if you don't put `#include <soci/soci.h>` before others soci headers build fails because of a type definition lack. It also forces me to put this header in a separate line (otherwise clang-format may try to rearrange lines).
I would suggest to leave this as is.